modules/core/src/main/java/org/apache/ignite/internal/cluster/DistributedTransactionConfiguration.java

```diff
@@ -0,0 +1,328 @@
+                    }
+                }
+            }
+        );
+    }
+
+    /**
+     * Cluster wide update of {@link #longOperationsDumpTimeout}.
+     *
+     * @param timeout New value of {@link #longOperationsDumpTimeout}.
+     * @return Future for {@link #longOperationsDumpTimeout} update operation.
+     * @throws IgniteCheckedException If failed during cluster wide update.
+     */
+    public GridFutureAdapter<?> updateLongOperationsDumpTimeoutAsync(long timeout) throws IgniteCheckedException {
+        return longOperationsDumpTimeout.propagateAsync(timeout);
+    }
+
+    /**
+     * Local update of {@link #longOperationsDumpTimeout}.
+     *
+     * @param timeout New value of {@link #longOperationsDumpTimeout}.
+     */
+    public void updateLongOperationsDumpTimeoutLocal(long timeout) {
+        longOperationsDumpTimeout.localUpdate(timeout);
+    }
+
+    /**
+     * @return Long operations dump timeout.
+     */
+    public Long longOperationsDumpTimeout() {
+        return longOperationsDumpTimeout.getOrDefault(dfltLongOpsDumpTimeout);
+    }
+
+    /**
+     * Cluster wide update of {@link #longTransactionTimeDumpThreshold}.
+     *
+     * @param longTransactionTimeDumpThreshold New value of threshold timeout in milliseconds.
+     * @return Future for {@link #longTransactionTimeDumpThreshold} update operation.
+     * @throws IgniteCheckedException If failed during cluster wide update.
+     */
+    public GridFutureAdapter<?> updateLongTransactionTimeDumpThresholdAsync(long longTransactionTimeDumpThreshold) throws IgniteCheckedException {
+        return this.longTransactionTimeDumpThreshold.propagateAsync(longTransactionTimeDumpThreshold);
+    }
+
+    /**
+     * Local update of {@link #longTransactionTimeDumpThreshold}.
+     *
+     * @param longTransactionTimeDumpThreshold New value of {@link #longTransactionTimeDumpThreshold}.
+     */
+    public void updateLongTransactionTimeDumpThresholdLocal(long longTransactionTimeDumpThreshold) {
+        this.longTransactionTimeDumpThreshold.localUpdate(longTransactionTimeDumpThreshold);
+    }
+
+    /**
+     * @return Threshold timeout for long transactions.
+     */
+    public Long longTransactionTimeDumpThreshold() {
+        return longTransactionTimeDumpThreshold.getOrDefault(dfltLongTransactionTimeDumpThreshold);
+    }
+
+    /**
+     * Cluster wide update of {@link #transactionTimeDumpSamplesCoefficient}.
+     *
+     * @param transactionTimeDumpSamplesCoefficient New coefficient for samples of completed transactions that will be dumped in log.
+     * @return Future for {@link #transactionTimeDumpSamplesCoefficient} update operation.
+     * @throws IgniteCheckedException If failed during cluster wide update.
+     */
+    public GridFutureAdapter<?> updateTransactionTimeDumpSamplesCoefficientAsync(double transactionTimeDumpSamplesCoefficient) throws IgniteCheckedException {
```

Comment:
Here and below: too long line. Personally, I'm ok with it but formally it's a code style violation